Update STYLE.md to expand API Page Link options (#384)
Roblox/creator-docs: STYLE.md

- Use present tense whenever possible, even for things that happen after something else.

  - **BAD**: You pressed the button, so the cursor will change soon.
  - **BETTER**: After you press the button, the cursor changes.

- Use active voice whenever possible. Active voice forces you to include the subject, which makes an enormous difference in clarity.

>   - **WORSE**: The list of players is returned.

Author: ```suggestion
  - **BAD**: The list of players is returned.
```

  - **BETTER**: The server returns the list of players.

- The terms "we" and "our" implies Roblox and is usually reserved for recommendations.

  - **BAD**: Now that we're happy with the terrain, we can start placing our models.
  - **BETTER**: When you're satisfied with the terrain, begin placing your models.

- Use second-person to address the reader, but try to avoid saying "you" (and especially "you can") too often.

>   - **WORSE**: If you want to build a custom user interface, you can add a container.

Author: ```suggestion
  - **BAD**: If you want to build a custom user interface, you can add a container.
```

  - **BETTER**: Add a container to begin building a custom user interface.

- Use **bold** for key terms and UI elements, **not** italics. Use `monospace` for file paths, code, functions, variables, etc.

  - **BAD**: It's _very important_ that you click the stop button after running **./script.sh**.
  - **BETTER**: Click the **Stop** button after running `./script.sh`.

- Avoid idioms or colloquialisms. They're hard to localize and confuse many readers.

>   - **WORSE**: Publishing in Roblox is a piece of cake.

Author: ```suggestion
  - **BAD**: Publishing in Roblox is a piece of cake.
```

  - **BETTER**: Publishing in Roblox is fast and simple.

- "Can" is for optional actions or permission, "might" for possibility, "must" for requirements, and "should" for recommendations. Avoid "may."

>   - **WORSE**: You should accept the license agreement before you may make changes. You might also review the coding standards.

Author: ```suggestion
  - **BAD**: You should accept the license agreement before you may make changes. You might also review the coding standards.
```

  - **BETTER**: You must accept the license agreement before you can make changes. You should also review the coding standards.

- Avoid gender identifiers. Whenever possible, make subjects plural.

  - **BAD**: After clicking, a player can see his or her inventory on his/her screen.
  - **BETTER**: After clicking, players can see their inventories on screen.

- Use "select" for options or objects in an experience. Use "click" for buttons or icons.

  - **BAD**: Click and drag to grab the signpost. Then select the **Color** button.
  - **BETTER**: Select the signpost and click the **Color** button.

- Spell out one (1) through nine (9). Start using numerals at 10. If a number needs a unit (GB, pounds, millimeters, kg, celsius, etc.), always use numerals, even if the number if smaller than 10.

  - **BAD**: 3 people looked for thirteen files on a six GB hard drive.
  - **BETTER**: Three people looked for 13 files on a 6 GB hard drive.

## Links

> ### Off-Site Links

Author: ```suggestion
### External Links
```

> For links to other domains or the web, use standard Markdown instead of JSX or HTML:

Author: ```suggestion
For links to other domains or the web outside of [create.roblox.com/docs](https://create.roblox.com/docs), use standard Markdown instead of JSX or HTML:
```

- `[Creator Dashboard](https://create.roblox.com/creations)`
- `[Blender](https://www.blender.org/)`

### Documentation Site Links

When linking to another page on the Roblox Documentation site, use **relative** links to the Markdown file, including the `.md` extension:

- `[Meshes](../parts/meshes.md)`
- `[Merch Booth](../resources/modules/merch-booth.md)`

> ### Engine API Links

Author: ```suggestion
### Engine API Reference Page Links
```

> When referencing a Roblox Engine API class, method, etc. in prose, use [automatic API links](#link-syntax). Do **not** use HTML or Markdown links to a specific URL such as `[BasePart](https://create.roblox.com/docs/reference/engine/classes/BasePart.html)`.

Author: ```suggestion
When referencing a Roblox Engine API class, method, etc. in prose, use our special [API link syntax](#link-syntax). Do **not** use HTML or Markdown links to a specific URL such as `[BasePart](https://create.roblox.com/docs/reference/engine/classes/BasePart.html)`.
```

#### Usage Guidelines

> - In prose, you should always link to an API member, but avoid excessive usage through pronouns and contextual references. For example:

Author: ```suggestion
- In prose, you should always link to an API member using our [API link syntax](#api-link-syntax), but avoid excessive usage through pronouns and contextual references. For example:
```

>   - **WORSE**: [`BasePart`](https://create.roblox.com/docs/reference/engine/classes/BasePart) is an abstract base class for in-world objects that render and are physically simulated. [`BaseParts`](https://create.roblox.com/docs/reference/engine/classes/BasePart) can be grouped into simulated rigid assemblies, and [`BaseParts`](https://create.roblox.com/docs/reference/engine/classes/BasePart) can detect collisions with other [`BaseParts`](https://create.roblox.com/docs/reference/engine/classes/BasePart).

Author: ```suggestion
  - **BAD**: [`BasePart`](https://create.roblox.com/docs/reference/engine/classes/BasePart) is an abstract base class for in-world objects that render and are physically simulated. [`BaseParts`](https://create.roblox.com/docs/reference/engine/classes/BasePart) can be grouped into simulated rigid assemblies, and [`BaseParts`](https://create.roblox.com/docs/reference/engine/classes/BasePart) can detect collisions with other [`BaseParts`](https://create.roblox.com/docs/reference/engine/classes/BasePart).
```

  - **BETTER**: The color of a [`BasePart`](https://create.roblox.com/docs/reference/engine/classes/BasePart) is defined by its [`Color`](https://create.roblox.com/docs/reference/engine/classes/BasePart#Color) property.

- If a term should be **plural** in prose, append the plural term after a `|` as outlined in [Link Substitution](#link-substitution).

  - **BAD**: A model can contain several [`BasePart`](https://create.roblox.com/docs/reference/engine/classes/BasePart)s.
  - **BETTER**: A model can contain several [`BaseParts`](https://create.roblox.com/docs/reference/engine/classes/BasePart).

- For words that are both **concepts** and API **terms** (for example, a [`Touched`](https://create.roblox.com/docs/reference/engine/classes/BasePart#Touched) event), consider whether you're referring to the concept or the term. Concepts require no special formatting.

  - **API TERM**: Connect the function to the [`Touched`](https://create.roblox.com/docs/reference/engine/classes/BasePart#Touched) event.
  - **CONCEPT**: Display text after a player's character has touched the part.

> #### Link Syntax

Author: ```suggestion
#### API Link Syntax
```

> To link to a page in the Roblox Engine API, use the following syntax patterns. The parser detects these keywords and forms monospaced links on the API page:

Author: ```suggestion
To link to a page in the Roblox Engine API, use the following syntax patterns. The parser detects these keywords and forms monospaced links to the proper Engine API page in our site for you.
```

[Meshes](../parts/meshes.md)
```

To link to a page in the Roblox Engine API, use the following syntax. The parser detects these keywords and creates monospaced links to the API page:

- `` `Class.Name` ``
- `` `Class.Name.Property` ``
- `` `Class.Name:Method()` ``
- `` `Class.Name:Method(argument)` ``
- `` `Class.Name:Method(arg1, arg2, arg3, ...)` ``
- `` `Class.Name.Event` ``
- `` `Class.Name.Callback` ``

- `` `Datatype.Name` ``
- `` `Datatype.Name:Method()` ``
- `` `Datatype.Name:Method(argument)` ``
- `` `Datatype.Name:Method(arg1, arg2, arg3, ...)` ``
- `` `Datatype.Name.constructor()` ``
- `` `Datatype.Name.constructor(argument)` ``
- `` `Datatype.Name.constructor(arg1, arg2, arg3, ...)` ``

- `` `Enum.Name` ``

> - `` `Global.LuaGlobals.Name` ``
> - `` `Global.RobloxGlobals.Name` ``

Reviewer: ```suggestion
- `` `Global.LuaGlobals.Function()` ``
- `` `Global.RobloxGlobals.Function()` ``
```

> - `` `Global.RobloxGlobals.Function()` ``
> - `` `Library.Name` ``

Reviewer: ```suggestion
- `` `Global.RobloxGlobals.Function()` ``
- `` `Global.RobloxGlobals.Property` ``
- `` `Library.Name` ``
```

> - `` `Library.Name.function()` ``

Reviewer: ```suggestion
- `` `Library.Name.Function()` ``
```

> - `` `Library.Name.function(argument)` ``
> - `` `Library.Name.function(arg1, arg2, arg3, ...)` ``

Reviewer: ```suggestion
- `` `Library.Name.Function(argument)` ``
- `` `Library.Name.Function(arg1, arg2, arg3, ...)` ``
```

> The default link text will display without the initial API type prefix (`Class.`, `Datatype.`, `Enum.`, `Global.`, `Library.`). For example, `` `Class.BasePart` `` will render as simply `BasePart`.

Reviewer: ```suggestion
The default link text will display without the initial API type prefix (`Class.`, `Datatype.`, `Enum.`, `Global.`, `Library.`). For example, `` `Class.BasePart` `` will render as simply `Class.BasePart`.
```

[Meshes](../parts/meshes.md)
```

To link to a page in the Roblox Engine API, use the following syntax. The parser detects these keywords and creates monospaced links to the API page:

- `` `Class.Name` ``
- `` `Class.Name.Property` ``
- `` `Class.Name:Method()` ``
- `` `Class.Name:Method(argument)` ``
- `` `Class.Name:Method(arg1, arg2, arg3, ...)` ``
- `` `Class.Name.Event` ``
- `` `Class.Name.Callback` ``

- `` `Datatype.Name` ``
- `` `Datatype.Name:Method()` ``
- `` `Datatype.Name:Method(argument)` ``
- `` `Datatype.Name:Method(arg1, arg2, arg3, ...)` ``
- `` `Datatype.Name.constructor()` ``
- `` `Datatype.Name.constructor(argument)` ``
- `` `Datatype.Name.constructor(arg1, arg2, arg3, ...)` ``

- `` `Enum.Name` ``

- `` `Global.LuaGlobals.Function()` ``
- `` `Global.RobloxGlobals.Function()` ``
- `` `Global.RobloxGlobals.Property` ``
- `` `Library.Name` ``
- `` `Library.Name.Function()` ``
- `` `Library.Name.Function(argument)` ``
- `` `Library.Name.Function(arg1, arg2, arg3, ...)` ``

> The default link text will display without the initial API type prefix (`Class.`, `Datatype.`, `Enum.`, `Global.`, `Library.`). For example, `` `Class.BasePart` `` will render as simply `Class.BasePart`.

Reviewer: ```suggestion
The default link text will display without the initial API type prefix (`Class.`, `Datatype.`, `Enum.`, `Global.`, `Library.`). For example, `` `Class.BasePart` `` will render as simply <a href="https://create.roblox.com/docs/reference/engine/classes/BasePart">BasePart</a>.
```

[Meshes](../parts/meshes.md)
```

To link to a page in the Roblox Engine API, use the following syntax. The parser detects these keywords and creates monospaced links to the API page:

- `` `Class.Name` ``
- `` `Class.Name.Property` ``
- `` `Class.Name:Method()` ``
- `` `Class.Name:Method(argument)` ``
- `` `Class.Name:Method(arg1, arg2, arg3, ...)` ``
- `` `Class.Name.Event` ``
- `` `Class.Name.Callback` ``

- `` `Datatype.Name` ``
- `` `Datatype.Name:Method()` ``
- `` `Datatype.Name:Method(argument)` ``
- `` `Datatype.Name:Method(arg1, arg2, arg3, ...)` ``
- `` `Datatype.Name.constructor()` ``
- `` `Datatype.Name.constructor(argument)` ``
- `` `Datatype.Name.constructor(arg1, arg2, arg3, ...)` ``

- `` `Enum.Name` ``

- `` `Global.LuaGlobals.Name` ``
- `` `Global.RobloxGlobals.Name` ``
- `` `Library.Name` ``
- `` `Library.Name.function()` ``
- `` `Library.Name.function(argument)` ``
- `` `Library.Name.function(arg1, arg2, arg3, ...)` ``

> The default link text will display without the initial API type prefix (Class, Datatype, Enum, Global, Library).

Reviewer: ```suggestion
The default link text will display without the initial API type prefix (`Class.`, `Datatype.`, `Enum.`, `Global.`, `Library.`). For example, `` `Class.BasePart` `` will render as simply `Class.BasePart`.
```

Reviewer: I think this will render out properly once viewed? Need to confirm though, that my usage of backticks will do what I want it to...

If desired, override the default link text using a `|` character:

- `` `Class.Name.Property|PropertyName` ``
- `` `Enum.Name|EnumItemName` ``

> - `` `Enum.Name|Custom text here` ``

Reviewer: ```suggestion
```

Reviewer: I don't think I've seen any cases where we'd want to use fairly arbitrary text in place of an Enum item name.

      <td><code>`Library.coroutine.create()`</code></td>
      <td><code><a href="https://create.roblox.com/docs/reference/engine/libraries/coroutine">coroutine.create()</a></code></td>
    </tr>
   <tr>

>       <td><code>`Class.Beam:SetTextureOffset(1)|myBeam:SetTextureOffset(1)`</code></td>

Reviewer: ```suggestion
      <td><code>`Class.Beam:SetTextureOffset()|myBeam:SetTextureOffset(1)`</code></td>
```

      <td><code>`Library.coroutine.create()`</code></td>
      <td><code><a href="https://create.roblox.com/docs/reference/engine/libraries/coroutine">coroutine.create()</a></code></td>
    </tr>
   <tr>
      <td><code>`Class.Beam:SetTextureOffset()|myBeam:SetTextureOffset(1)`</code></td>
      <td><code><a href="https://create.roblox.com/docs/reference/engine/classes/Beam#SetTextureOffset">myBeam:SetTextureOffset(1)</a></code></td>
    </tr>
    <tr>
      <td><code>`Datatype.CFrame.new(x, y, z)`</code></td>
      <td><code><a href="https://create.roblox.com/docs/reference/engine/datatypes/CFrame#new">CFrame.new(x, y, z)</a></code></td>
    </tr>
  </tbody>
</table>

#### Link Substitution

If desired, you can override the default link text using a `|` character followed by the desired term:

- `` `Class.Name.Property|PropertyName` ``
- `` `Enum.Name|EnumItemName` ``

This technique should be used to shorten the length of rendered links on the page:

> - **WORSE**: [`MarketplaceService`](https://create.roblox.com/docs/reference/engine/classes/MarketplaceService) is responsible for in-experience transactions. The most notable methods are [`MarketplaceService:PromptProductPurchase()`](https://create.roblox.com/docs/reference/engine/classes/MarketplaceService#PromptProductPurchase) and [`MarketplaceService:PromptPurchase()`](https://create.roblox.com/docs/reference/engine/classes/MarketplaceService#PromptPurchase), as well as the callback [`MarketplaceService.ProcessReceipt`](https://create.roblox.com/docs/reference/engine/classes/MarketplaceService#ProcessReceipt) which must be defined so that transactions do not fail.

Author: ```suggestion
- **BAD**: [`MarketplaceService`](https://create.roblox.com/docs/reference/engine/classes/MarketplaceService) is responsible for in-experience transactions. The most notable methods are [`MarketplaceService:PromptProductPurchase()`](https://create.roblox.com/docs/reference/engine/classes/MarketplaceService#PromptProductPurchase) and [`MarketplaceService:PromptPurchase()`](https://create.roblox.com/docs/reference/engine/classes/MarketplaceService#PromptPurchase), as well as the callback [`MarketplaceService.ProcessReceipt`](https://create.roblox.com/docs/reference/engine/classes/MarketplaceService#ProcessReceipt) which must be defined so that transactions do not fail.
```

- **BETTER**: [`MarketplaceService`](https://create.roblox.com/docs/reference/engine/classes/MarketplaceService) is responsible for in-experience transactions. The most notable methods are [`PromptProductPurchase()`](https://create.roblox.com/docs/reference/engine/classes/MarketplaceService#PromptProductPurchase) and [`PromptPurchase()`](https://create.roblox.com/docs/reference/engine/classes/MarketplaceService#PromptPurchase), as well as the callback [`ProcessReceipt`](https://create.roblox.com/docs/reference/engine/classes/MarketplaceService#ProcessReceipt) which must be defined so that transactions do not fail.

You should also use link substitution when framing an API term in plural:

- **BAD**: A model can contain several [`BasePart`](https://create.roblox.com/docs/reference/engine/classes/BasePart)s.

> - **BETTER**: A model can contain several [`BaseParts`](https://create.roblox.com/docs/reference/engine/classes/BasePart).

Author: I may suggest "Better" being replaced with the opposite of **BAD**, using "`**GOOD**`" instead
